modules/pagerduty/pagerduty.go

```diff
@@ -0,0 +1,188 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Miles Crabill <mcrabill@mozilla.com>
+
+package pagerduty
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	pagerduty "github.com/PagerDuty/go-pagerduty"
+
+	"go.mozilla.org/userplex/modules"
+)
+
+func init() {
+	modules.Register("pagerduty", new(module))
+}
+
+type module struct{}
+
+func (m *module) NewRun(c modules.Configuration) modules.Runner {
+	r := new(run)
+	r.Conf = c
+	return r
+}
+
+type run struct {
+	Conf modules.Configuration
+	p    parameters
+	c    credentials
+
+	pagerdutyEmailToLdapEmail  map[string]string
+	ldapEmailToPagerdutyEmails map[string][]string
+}
+
+type parameters struct {
+	Subdomain string
+}
+
+type credentials struct {
+	APIKey string
+}
+
+func (r *run) Run() (err error) {
+	err = r.Conf.GetParameters(&r.p)
+	if err != nil {
+		return
+	}
+	err = r.Conf.GetCredentials(&r.c)
+	if err != nil {
+		return
+	}
+
+	r.buildLdapMapping()
+	ldapers := r.getLdaperEmails()
+
+	client := pagerduty.NewClient(r.c.APIKey)
+
+	var allUsers []pagerduty.User
+	offset := uint(0)
+	opt := pagerduty.ListUserOptions{
+		APIListObject: pagerduty.APIListObject{Offset: offset},
+	}
+	for {
+		usersResp, err := client.ListUsers(opt)
+		if err != nil {
+			log.Fatalf("[error] pagerduty %s: %s", r.p.Subdomain, err.Error())
+		}
+		allUsers = append(allUsers, usersResp.Users...)
+		if !usersResp.More {
+			break
+		}
+		offset += usersResp.Limit
+		opt.APIListObject.Offset = offset
+	}
+
+	log.Printf("[info] pagerduty %s: found %d users", r.p.Subdomain, len(allUsers))
+
+	// all pagerduty users
+	for _, user := range allUsers {
+		// if there's an ldapuid/localuid mapping for the user
+		var ldapEmail, ldapEmailString string
```

Comment:
this variable name does not tell me what an `ldapEmailString` is and how it differs from `ldapEmail`. I am puzzled.